src/profiler.cpp

```diff
@@ -1731,6 +1732,26 @@ void Profiler::timerLoop(void* timer_id) {
     }
 }
 
+void Profiler::logSampleCollectionStatus(Arguments& args, u64 printed_samples_count, Writer& out) {
+    if (!out.good()) {
+        Log::warn("Output file may be incomplete");
+    }
+    // Skip logging if profiling is invoked with --loop mode
+    if (args._loop) {
+        return;
+    }
+    // If total samples count match the samples which were skipped - we log a message
```

Comment:
Please avoid comments that just [repeat](https://www.reddit.com/r/ProgrammerHumor/comments/i4v6b2/comments/) what is written in the code.
IMO, in this case, the code is pretty much self-describing. I'd keep it concise like
```
    if (_total_samples == _failures[-ticks_skipped]) {
        Log::info("No samples were collected");
    } else if (printed_samples_count == 0) {
        Log::info("All samples were filtered out");
    }
```

Reply:
Btw, although I typically prefer simpler expressions, in this particular case
`if (_total_samples - _failures[-ticks_skipped] == 0)` would be more obvious to a reader (i.e. "No samples" means that some value is zero).